packages/devtools_server/lib/src/memory_profile.dart

```diff
@@ -245,27 +247,27 @@ class MemoryProfile {
   /// Call to returns JSON payload 'EstimateRasterCacheMemory' with two entries:
   ///   layerBytes - layer raster cache entries in bytes
   ///   pictureBytes - picture raster cache entries in bytes
-  Future<Response> getRasterCacheMetrics(IsolateRef selectedIsolate) async {
+  Future<Response?> getRasterCacheMetrics(IsolateRef selectedIsolate) async {
     final viewId = await getFlutterViewId(selectedIsolate);
 
     return await callService(
       registrations.flutterEngineRasterCache,
-      args: <String, String>{
+      args: <String, String?>{
         'viewId': viewId,
       },
       isolateId: selectedIsolate.id,
     );
   }
 
-  void _update(VM vm, List<Isolate> isolates) {
-    processRss = vm.json['_currentRSS'];
+  void _update(VM vm, List<Isolate?> isolates) {
+    processRss = vm.json!['_currentRSS'];
 
     isolateHeaps.clear();
 
-    for (Isolate isolate in isolates) {
+    for (Isolate? isolate in isolates) {
       if (isolate != null) {
-        final List<HeapSpace> heaps = getHeaps(isolate).toList();
-        isolateHeaps[isolate.id] = heaps;
+        final List<HeapSpace?> heaps = getHeaps(isolate).toList();
+        isolateHeaps[isolate.id!] = heaps as List<HeapSpace>;
```

Comment:
this isn't right.  For this to make sense, getHeaps should just return a List<HeapSpace>
I would be worried this code as written will just throw errors at runtime.

Reply:
fixed getHeaps to exclude null HeapSpaces returned from HeapSpace.parse